packages/mural-picker/src/components/mural-picker/index.tsx

```diff
@@ -307,34 +306,34 @@ const MuralPicker = ({
     setError('');
     setRoom(null);
     setViewType(tab);
+    // add mural to Recent murals and select it
+    if (newMural) {
+      // recent tab to display new mural first and select it
+      setRecentMurals([newMural.mural, ...muralType[tab]]);
+      setMurals([newMural.mural, ...muralType[tab]]);
+      setMural(newMural.mural);
+      onSelect(newMural.mural, newMural.room, newMural.workspace!);
+    }
     // mural tab data exists
     if (muralType[tab].length > 0) {
-      // add mural to Recent murals and select it
-      if (newMural) {
-        // recent tab to display new mural first and select it
-        setRecentMurals([newMural.mural, ...muralType[tab]]);
-        setMurals([newMural.mural, ...muralType[tab]]);
-        setMural(newMural.mural);
-        onSelect(newMural.mural, newMural.room, newMural.workspace!);
-      } else {
-        setMurals(muralType[tab]);
-        if (tab === ViewType.ALL) {
-          setWorkspace(defaultWorkspace);
-        }
+      setMurals(muralType[tab]);
```

Comment:
If `newMural` is not undefined, `setMurals` will be call here and at line 313. I'm wondering if this can cause any unwanted side effect.

Reply:
Oh good point! I meant to make that an else if